### PR TITLE
Change how spaces in component names and attributes are handled

### PR DESCRIPTION
**Pull request by jisqyv**
Add back spaces into the French Translation. Yail generation should
now always use the english names of component properties, methods and
events. So spaces or other characters that are not legal Java
identifiers can now appear in translations of these.

Component types also can now have spaces. When a component is dragged
onto the designer, spaces and single quotes are now converted to
underscores.

Change-Id: I03e49588528c06b6f853a5ef5bfe5ae11266a51e

### CODE REVIEW

**Comment by jisqyv** [comment]
@afmckinney @halatmit This code is on ai2-test and Adel Kassah (who did the French translation and original stumbled on this issue) says it works with his test cases.

File: appinventor/blocklyeditor/src/generators/yail.js [hunk 1]
```diff
@@ -613,4 +613,25 @@ Blockly.Yail.blockToCode1 = function(block) {
   }
 };
 
-
+Blockly.Yail.encodeName = function(name) {
```
Comment (halatmit) [inline appinventor/blocklyeditor/src/generators/yail.js]: Please add a comment to explain what this is for.

**Comment by jisqyv** [comment]
@halatmit Actually, I'm not using that routine anymore. I flushed it.

**Comment by halatmit** [comment]
LGTM

**Comment by jisqyv** [comment]
rebased

**Pull request merged**